### ISSUE TITLE
./bin/www issue with github commit and deployment

### ISSUE THREAD

**Issue by vickyrathee** [expressjs/generator#261]
Expressjs default template has `.bin/www` to start the node js app. But `bin` folder is never committed to github (I can't find anyway until now).

So my codeCommit pipeline fails -

```
-------------------------------------
/var/log/nodejs/nodejs.log
-------------------------------------
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! myApp@0.0.0 start: `node ./bin/www`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the myApp@0.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

```

![image](https://user-images.githubusercontent.com/6106479/80917251-65678b80-8d7b-11ea-9f3b-697fd1f46a12.png)

Should I delete the bin folder and rename www with `server.js` ? What exactly the `www` does? If I put it on main directory with rename..Will it cause any problem?

Or anyway to push it on git commit?

**Comment by dougwilson** [comment]
If it is not being added with `get add bin/www` it sounds like you may have a git ignore ignoring that directory or file. You can generate your project with `--git` option to create a `.gitignore` with the expected filters.

**Comment by vickyrathee** [comment]
No, it's nowhere in .gitignore. I saw other people have similar issue and can't upload `/bin` folder https://stackoverflow.com/questions/27575228/cant-push-bin-directory-to-github

**Comment by ryhinchey** [comment]
@vickyrathee Do you happen to have `bin` in a global gitignore file?

**Comment by jonchurch** [comment]
You can rule out if it's a a gitignore issue with a few commands.

Check what files are currently being ignored with:

```
git status --ignored
```

And then find what gitignore is responsible with:

```
git check-ignore -v .bin
```
